### DISCUSSION TITLE
3d ultrasound simulation

### DISCUSSION THREAD

**Discussion by pranavajy** [mlavik1/UnityVolumeRendering#224] [Q&A]
Hey,  is there any way i can use this plugin for 3d ultrasound simulation? Your help would be much appreciated : )

**Comment by pranavajy** [comment]
my project is simulating third trimester  fetal ultrasound  scan in vr.

**Comment by mlavik1** [comment]
Hi!
Good question!
I unfortunately don't have much experience with 3D ultrasound. I received one 3D ultrasound dataset once, which only consisted of 3 DICOM slices and could be rendered really well.
The plugin can basically render any volumetric dataset that's represented by a 3-dimensional array of density values. Supported formats are DICOM, NRRD, NIFTI, PARCGH, image sequences and raw files. If your dataset is in one of these formats then it should in general be possible to render it, although there are no advanced features for rendering 3D ultrasound datasets specifically.

**Comment by pranavajy** [comment]
what about creating a 2d ultrasound simulator with sequence of images ? any thoughts? my aim is to create an ultrasound simulator with  dynamic view based on the positioning of controller.

**Comment by mlavik1** [comment]
If you just want to display 2D slices, then I suppose you could just use SImpleITK to load the data files (DICOM?) and convert them to 2D textures. Alternatively, [fo-dicom](https://github.com/fo-dicom/fo-dicom) might be a bit easier to use, and more documented for usage in Unity.
Again, I don't have any experience with ultrasound data - so I might be missing something😅
